File: backend/src/main/java/de/schnellertermin/backend/services/abstracts/CategoryService.java
package de.schnellertermin.backend.services.abstracts;

import de.schnellertermin.backend.services.dtos.requests.CategoryRequest;
import de.schnellertermin.backend.services.dtos.responses.CategoryCreatedResponse;
import de.schnellertermin.backend.services.dtos.responses.CategoryGetAllResponse;

import java.util.List;

public interface CategoryService {
    List<CategoryGetAllResponse> getAllCategories();
    CategoryCreatedResponse getCategoryById(String id);
    CategoryCreatedResponse addCategory (CategoryRequest categoryRequest);
    CategoryCreatedResponse updateCategory (String id, CategoryRequest categoryRequest);
    void deleteCategory (String id);
}

File: frontend/src/services/CategoryService.ts
import axios from "axios";
import {Category} from "../types/Category.ts";

export default class CategoryService {
    getAllCategories() {
        return axios.get('/api/categories');
    }

    getCategoryById(id: string) {
        return axios.get(`/api/categories/${id}`)
    }

    addCategory(category: Category) {
        return axios.post('/api/categories', category);
    }

    updateCategory(category: Category) {
        return axios.put('/api/categories/', category)
    }

    deleteCategory(id: number) {
        return axios.delete(`/api/categories/delete/${id}`)
    }
}